Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp

```diff
@@ -772,31 +772,21 @@ int CLuaEngineDefs::EngineSetModelLODDistance(lua_State* luaVM)
     return 1;
 }
 
-int CLuaEngineDefs::EngineResetModelLODDistance(lua_State* luaVM)
+//  bool engineResetModelLODDistance(int / string modelID)
+bool CLuaEngineDefs::EngineResetModelLODDistance(const std::variant<const std::string, const unsigned short> variantModelID)
 {
-    SString          strModel = "";
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadString(strModel);
+    const unsigned short modelID = std::holds_alternative<std::string>(variantModelID) ? CModelNames::ResolveModelID(std::get<std::string>(variantModelID)) : std::get<unsigned int>(variantModelID);
+    CModelInfo* const modelInfo = g_pGame->GetModelInfo(modelID);
+    if (!modelInfo)
+        throw std::invalid_argument(SString("Invalid model id %u", modelID));
 
-    if (argStream.HasErrors())
-        return luaL_error(luaVM, argStream.GetFullErrorMessage());
-    
-    unsigned short usModelID = CModelNames::ResolveModelID(strModel);
-    CModelInfo*    pModelInfo = g_pGame->GetModelInfo(usModelID);
-    if (pModelInfo)
-    {
-        float fCurrentDistance = pModelInfo->GetLODDistance();
-        float fOriginalDistance = pModelInfo->GetOriginalLODDistance();
-        //Make sure we're dealing with a valid LOD distance, and not setting the same LOD distance
-        if (fOriginalDistance > 0.0f && fOriginalDistance != fCurrentDistance) {
-            pModelInfo->SetLODDistance(fOriginalDistance, true);
-            lua_pushboolean(luaVM, true);
-            return 1;
-        }
-    }
+    //Make sure we're dealing with a valid LOD distance, and not setting the same LOD distance
```

Comment:
EngineResetModelLODDistance
```suggestion
    // Make sure we're dealing with a valid LOD distance, and not setting the same LOD distance
```